site/specs/webrtc.yml

```diff
@@ -2421,6 +2520,201 @@ components:
+          description: the name of the state that the Dialog has entered
+          example: ANSWERED
+        tag:
+          type: string
+          description: >-
+            User defined tag to associate with the dialog.  Must not contain
+            PII.
+          example: sip-patron-dialog
+    eventBody:
+      oneOf:
+        - $ref: '#/components/schemas/sipDialogEventBody'
+        - $ref: '#/components/schemas/onLeaveEventBody'
+        - $ref: '#/components/schemas/onDisconnectEventBody'
+        - $ref: '#/components/schemas/onConnectEventBody'
+      discriminator:
+        propertyName: event
+        mapping:
+          onSipDialogStateChange: '#/components/schemas/sipDialogEventBody'
+          onLeave: '#/components/schemas/onLeaveEventBody'
+          onDisconnect: '#/components/schemas/onDisconnectEventBody'
+          onConnect: '#/components/schemas/onConnectEventBody'
+    baseEventBody:
+      type: object
+      description: the common properties shared by all events
+      required:
+        - event
+      properties:
+        event:
+          type: string
+          description: >-
+            the name of the event, and the condition that describes the body
+            that will be present
+          example: onSipDialogStateChange
+          enum:
+            - onSipDialogStateChange
+            - onLeave
+            - onDisconnect
+            - onConnect
+        timestamp:
+          type: string
+          description: the epoch UTC time, in milliseconds, when the event happened
+          example: '1628786234676'
+        participantId:
+          type: string
+          description: the ID of Participant on which the event was detected
+          example: be199214-1a03-402a-a7b6-d002470a0465
+        dialogId:
+          type: string
+          description: the ID of Participant on which the event was detected
+          example: be199214-1a03-402a-a7b6-d002470a0465
+        tag:
+          type: string
+          description: >-
+            the tag associated with the Participant on which the event was
+            detected.
+          example: hello-world-browser
+      discriminator:
+        propertyName: event
+    dialogEventData:
+      type: object
+      description: >
+        The dialogEventData element contains information about the event that
+        triggered the transition to the  current Dialog state. This data is
+        dependent on extrinsic events that impact the Dialog. An understanding
+        of the details of the Dialog Event may affect the business logic choices
+        for the receiving application.
+      properties:
+        eventReason:
+          type: string
+          description: A summary of why the event happened
+          example: FAR END BUSY
+        eventErrorReason:
+          type: string
+          description: If the event was an error, a text description of the error
+          example: Destination Server unavailable
+        eventErrorCode:
```

Comment:
Is this intended to be the SIP status code? In the code that's what it is. 

If it is the SIP status code then the description should say so. If it represents something else we should point to a page where we list out the possible values. Otherwise the reader has no idea what this means

Reply:
I put the code in there because it is one pattern for providing error information in a payload that can be acted on by the end system.   I think that the best bet is a good convergent discussion.